be/src/formats/parquet/column_reader.cpp

```diff
@@ -194,37 +194,58 @@ class MapColumnReader : public ColumnReader {
             map_column = down_cast<vectorized::MapColumn*>(dst);
         }
         auto* key_column = map_column->keys_column().get();
-        // read key
-        auto st = _key_reader->prepare_batch(num_records, content_type, key_column);
-        if (!st.ok() && !st.is_end_of_file()) {
-            return st;
+        auto* value_column = map_column->values_column().get();
+        Status st;
+        if (_key_reader != nullptr) {
+            st = _key_reader->prepare_batch(num_records, content_type, key_column);
+            if (!st.ok() && !st.is_end_of_file()) {
+                return st;
+            }
         }
 
-        // read value
-        auto* value_column = map_column->values_column().get();
-        st = _value_reader->prepare_batch(num_records, content_type, value_column);
+        if (_value_reader != nullptr) {
+            st = _value_reader->prepare_batch(num_records, content_type, value_column);
+            if (!st.ok() && !st.is_end_of_file()) {
+                return st;
+            }
+        }
 
-        // check the value_column size is the same with key_column
-        DCHECK(value_column->size() == key_column->size());
+        if ((_key_reader != nullptr) & (_value_reader != nullptr)) {
+            // check the value_column size is the same with key_column
+            DCHECK(value_column->size() == key_column->size());
+        }
 
         level_t* def_levels = nullptr;
         level_t* rep_levels = nullptr;
         size_t num_levels = 0;
-        _key_reader->get_levels(&def_levels, &rep_levels, &num_levels);
+
+        if (_key_reader != nullptr) {
+            _key_reader->get_levels(&def_levels, &rep_levels, &num_levels);
+        } else {
+            _value_reader->get_levels(&def_levels, &rep_levels, &num_levels);
+        }
 
         auto& offsets = map_column->offsets_column()->get_data();
         offsets.resize(num_levels + 1);
         vectorized::NullColumn null_column(num_levels);
         auto& is_nulls = null_column.get_data();
         size_t num_offsets = 0;
         bool has_null = false;
-        // use key's def_levels, rep_levels, num_level to compute offset and nullable
+
         // ParquetFiled Map -> Map<Struct<key,value>>
         def_rep_to_offset(_field->children[0].level_info, def_levels, rep_levels, num_levels, &offsets[0], &is_nulls[0],
                           &num_offsets, &has_null);
         offsets.resize(num_offsets + 1);
         is_nulls.resize(num_offsets);
 
+        // fill with default
```

Comment:
If the field is not selected, do we need to fill it with default value?

Reply:
we will select at least one subfield, offset and null column will be filled with the selected field, so we only need fill default value for key or value column.